extra_scripts/create_imagenet_r_data_files.py

```diff
@@ -0,0 +1,276 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import argparse
+import importlib
+import os
+
+from torchvision.datasets.utils import download_and_extract_archive
+
+
+def get_argument_parser():
+    """
+    List of arguments supported by the script
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i",
+        "--input",
+        type=str,
+        help="Path to the folder containing the original imagenet-r folder",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=str,
+        help="Folder where the classification dataset will be written",
+    )
+    parser.add_argument(
+        "-d",
+        "--download",
+        action="store_const",
+        const=True,
+        default=False,
+        help="To download the original in the input folder",
+    )
+    return parser
+
+
+def download_dataset(root: str):
+    """
+    Download the Imagenet-R dataset archive and expand it in the folder
+    provided as parameter
+    """
+    URL = "https://people.eecs.berkeley.edu/~hendrycks/imagenet-r.tar"
+    download_and_extract_archive(url=URL, download_root=root)
+
+
+def import_imagenet_a_module():
+    script_path = os.path.split(__file__)[0]
+    return importlib.import_module("create_imagenet_a_data_files", script_path)
```

Comment:
I am not fully sure about this. Are we saying that we have a dependency on the other script ? Is it possible to expand the other script instead in that case and achieve `a` or `r` based on the user input?

Reply:
Yes, I can expand the other script instead.

I was merely trying to make sure that all scripts have the same interface (same options), but this clearly has a cost in terms of dependencies.

One option to keep the same uniform interface is to prepare both these datasets at the same time (and rename the script `create_imagenet_ood_test_splits.py`). What do you think?

Reply:
yes I think preparing at the same time and renaming as you suggested is great :)